app/scripts/metamask-controller.js

```diff
@@ -1018,15 +993,25 @@ export default class MetamaskController extends EventEmitter {
       removeIdentity: this.preferencesController.removeAddress.bind(
         this.preferencesController,
       ),
-      setAccountLabel: this.preferencesController.setAccountLabel.bind(
-        this.preferencesController,
-      ),
-      setSelectedAddress: this.preferencesController.setSelectedAddress.bind(
-        this.preferencesController,
-      ),
-      syncIdentities: this.preferencesController.syncAddresses.bind(
-        this.preferencesController,
-      ),
+      setAccountLabel: (address, label) => {
+        const accountToBeNamed =
+          this.accountsController.getAccountByAddress(address);
+        this.accountsController.setAccountName(accountToBeNamed.id, label);
+
+        this.preferencesController.setAccountLabel(address, label);
+      },
+      setSelectedAddress: (address) => {
+        const accountToBeSet =
+          this.accountsController.getAccountByAddress(address);
+        if (accountToBeSet) {
+          this.accountsController.setSelectedAccount(accountToBeSet.id);
+        }
```

Comment:
Do we want to throw an error if not found?

Reply:
Yes, done [957a205](https://github.com/MetaMask/metamask-extension/pull/21437/commits/957a205fe9e0a800c0b8bbff8ac3e314456ace88)